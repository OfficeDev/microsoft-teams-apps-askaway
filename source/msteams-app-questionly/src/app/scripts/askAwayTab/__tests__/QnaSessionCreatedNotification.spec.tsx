/**
 * @jest-environment jsdom
 */

// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import * as React from 'react';
import { shallow, configure } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import { QnaSessionCreatedNotification } from '../popups/QnaSessionCreatedNotification';
import QnaSessionNotificationInternal from '../popups/QnaSessionNotificationInternal';

configure({ adapter: new enzymeAdapterReact16() });

describe('QnaSessionCreatedNotification', () => {
    let url = '';
    beforeAll(() => {
        global.window = Object.create(window);
        url = 'https://dummy.com';
        Object.defineProperty(window, 'location', {
            value: {
                href: url,
            },
        });
    });

    it('should match the snapshot', () => {
        const wrapper = shallow(<QnaSessionCreatedNotification />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should render QnaSessionCreatedNotification', () => {
        const component = shallow(<QnaSessionCreatedNotification />);
        component.setState({ theme: {}, direction: 'ltr' });
        expect(component.find(QnaSessionNotificationInternal)).toHaveLength(1);
    });
});
