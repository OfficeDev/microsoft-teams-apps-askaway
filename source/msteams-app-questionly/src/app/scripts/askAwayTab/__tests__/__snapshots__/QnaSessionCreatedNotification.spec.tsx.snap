// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QnaSessionCreatedNotification should match the snapshot 1`] = `
<Provider
  style={
    Object {
      "background": "unset",
    }
  }
  theme={Object {}}
>
  <QnaSessionNotificationInternal
    onSubmitSession={[Function]}
    searchParams={
      URLSearchParams {
        Symbol(query): Array [],
        Symbol(context): "http://dummy.com/",
      }
    }
  />
</Provider>
`;
